consensus/src/consensus_observer/observer.rs

```diff
@@ -638,6 +694,18 @@ impl ConsensusObserver {
             return;
         };
 
+        // If all payloads exist, process the block. Otherwise, store it
+        // in the missing block store and wait for the payloads to arrive.
+        if self.all_payloads_exist(ordered_block.blocks()) {
+            self.process_ordered_block(ordered_block).await;
```

Comment:
does this work if the payload is out of order i.e block 1 is missing payload but block 2 has payload exists?

Reply:
Does my answer above make it clearer? If we try to process block 2 (because we already have a payload), we'll just drop it because it doesn't extend the chain, and fallback to state sync. Does this make sense?